### PR TITLE
adding vertical docs

### PR DESCRIPTION
**Pull request by melbarkerlbl**
First draft of .rst file for vertical transform and stretching.

### CODE REVIEW

**Comment by jmsexton03** [comment]
In order for it to be included in the html you'd need to add it to a specific part of the docs, similar to https://github.com/seahorce-scidac/ROMSX/blob/development/Docs/sphinx_doc/Discretizations.rst

you would also need to add it to index.rst as a file that gets loaded, maybe after Discretizations?
https://github.com/seahorce-scidac/ROMSX/blob/5b9432761548b09720f6ccc8c59cb54138a4bf1d/Docs/sphinx_doc/index.rst?plain=1#L60

File: Docs/sphinx_doc/VerticalSCoord.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+This transformation offers several advantages:
+
+* Regardless of the design of :math:'C\left( \sigma \right) ', it behaves like equally-spaced sigma-coordinates in shallow regions, where :math:'h\left( x,y\right) \ll h_c'. This is advantageous because it avoids excessive resolution and associated CFL limitation in such areas.
+
+* Near-surface refinement behaves more or less like geopotential coordinates in deep regions (level thicknesses, :math:'H_z', do not depend or weakly depend on bathymetry), while near-bottom like sigma coordinates (:math:'H_z' is roughly proportional to depth). This reduces the extreme r-factors near the bottom and reduces pressure gradient errors.
+
+* The **true** sigma-coordinate system is recovered as :math:'h_c \rightarrow \infty '. This is useful when configuring applications with **flat** bathymetry and **uniform** level thickness. Practically, you can achieve this by setting Tcline to **1.0d+16** in roms.in. This will set :math:'h_c =1.0\times 10^{16}'. Although not necessary, we also recommend that you set :math:'\theta _S = 0' and :math:'\theta _B =0'.
```
Comment (jmsexton03) [inline Docs/sphinx_doc/VerticalSCoord.rst]: Tcline and `roms.in` should be updated to refer to ROMSX variables `tcline` and `inputs` (although the value of tcline is currently not read from the inputs file it does exist in the DataStruct file)
https://github.com/seahorce-scidac/ROMSX/blob/5b9432761548b09720f6ccc8c59cb54138a4bf1d/Source/DataStruct.H#L335-L338

File: Docs/sphinx_doc/VerticalSCoord.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+   d\hat{z} =d\sigma h\left( x,y\right) \left[ \frac{h_c}{h_c +h\left( x,y\right) } \right]
+
+As a consequence, the uppermost grid box retains very little dependency from bathymetry in deep areas, where :math:'h_c \ll h\left( x,y\right) '. For example if :math:'h_c =250m' and :math:'h\left( x,y\right) ' changes from :math:'2000' to :math:'6000m', the uppermost grid box changes only by a factor of 1.08 (less than 10%).
+
+
+
+Vertical Stretching
+-------------------
+
+The above generic vertical transformation design facilitates numerous vertical stretching functions :math:'C\left( \sigma \right) '. This function is defined in terms of several parameters which are specified in the standard input file, roms.in.
```
Comment (jmsexton03) [inline Docs/sphinx_doc/VerticalSCoord.rst]: `roms.in` to `inputs` (see above)

File: Docs/sphinx_doc/VerticalSCoord.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+Vertical S-coordinate
+=====================
+
+.. _'ROMS documentation': https://www.myroms.org/wiki/Vertical_S-coordinate
+
+These equations are as given in the 'ROMS documentation'_
+
+ROMSX has a generalized vertical, terrain-following, coordinate system.  Currently, the vertical transformation equation :math:'z=z\left( x,y,\sigma ,t\right) ', is available which can support vertical stretching 1D-functions when several constraints are satisfied.
+
+Transformation Equation
```
Comment (jmsexton03) [inline Docs/sphinx_doc/VerticalSCoord.rst:14]: consider commenting on which transformation this corresponds to in ROMS (or if there's a name for it)

File: Docs/sphinx_doc/VerticalSCoord.rst [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+-----------------------
+
+.. math::
+   z\left( x,y,\sigma ,t\right) &=\zeta \left( x,y,t\right) +\left[ \zeta \left( x,y,t\right) +h\left( x,y\right) \right] S\left( x,y,\sigma \right) ,\\
+   S\left( x,y,\sigma \right) &=\frac{h_c \sigma +h \left( x,y\right) C\left( \sigma \right) }{h_c +h\left( x,y\right)}
+
+where :math:'S\left( x,y,\sigma \right)' is a nonlinear vertical transformation functional, :math:'\zeta \left( x,y,t\right)' is the time-varying free-surface, :math:'h\left( x,y\right) ' is the unperturbed water column thickness and :math:'z=-h\left( x,y\right) ' corresponds to the ocean bottom, :math:'\sigma ' is a fractional vertical stretching coordinate ranging from :math:'-1\leq \sigma \leq 0,C\left( \sigma \right) ' is a nondimensional, monotonic, vertical stretching function ranging from :math:'-1\leq C\left( \sigma \right) \leq 0', and :math:'h_c' is a positive thickness controlling the stretching.  In sediment applications, :math:'h=h\left( x,y,t\right) ' is changed at every time-step since it is affected by erosion and deposition processes.
```
Comment (jmsexton03) [inline Docs/sphinx_doc/VerticalSCoord.rst]: Good to have the full text, but maybe comment out the sediment part as that's not currently implemented in ROMSX

File: Docs/sphinx_doc/VerticalSCoord.rst [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+     \sigma \left( k \right) = \begin{cases}{\frac{k-N}{N}, & \text{at vertical }W\text{-points}, & k=0,\ldots ,N,}\\{\frac{k-N-0.5}{N}, & \text{at vertical}\rho \text{-points}, & k=1,\ldots ,N} \end{cases}
+
+* :math:'C\left( \sigma \right) ' must be constrained by :math:'-1 \leq C\left( \sigma \right) \leq 0', that is,
+
+  .. math::
+     C\left( \sigma \right) = \begin{cases}{0, & \text{if } \sigma = 0, & C\left( \sigma \right) = 0, & \text{at the free-surface};}\\{-1, & \text{if } \sigma = -1, & C\left( \sigma \right) = -1, & \text{at the ocean bottom}.} \end{cases}
+
+Available Stretching Functions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+4. A. Shchepetkin (2010) UCLA-ROMS current function, Vstretching=4. :math:'C\left( \sigma \right) ' is defined as a continuous, double stretching function:
```
Comment (jmsexton03) [inline Docs/sphinx_doc/VerticalSCoord.rst]: consider making this and the transformation numbering (and any references to ROMS numbering) consistent in this doc

File: Docs/sphinx_doc/VerticalSCoord.rst (hunk 1 above)
Comment (melbarkerlbl) [reply Docs/sphinx_doc/VerticalSCoord.rst]: perfect, that was one of my known issues

File: Docs/sphinx_doc/VerticalSCoord.rst (hunk 4 above)
Comment (melbarkerlbl) [reply Docs/sphinx_doc/VerticalSCoord.rst]: I wondered.

File: Docs/sphinx_doc/VerticalSCoord.rst (hunk 1 above)
Comment (melbarkerlbl) [reply Docs/sphinx_doc/VerticalSCoord.rst]: See if what it says now works?

File: Docs/sphinx_doc/VerticalSCoord.rst (hunk 2 above)
Comment (melbarkerlbl) [reply Docs/sphinx_doc/VerticalSCoord.rst]: Done

File: Docs/sphinx_doc/VerticalSCoord.rst (hunk 4 above)
Comment (melbarkerlbl) [reply Docs/sphinx_doc/VerticalSCoord.rst]: done

**Review by jmsexton03 — APPROVED** [review]

**Pull request merged**